src/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -504,8 +504,22 @@ internal static char ParseFormatSpecifier(string format, out int digits)
 
         private static string FormatBigIntegerToHex(BigInteger value, char format, int digits, NumberFormatInfo info)
         {
-            Debug.Assert(format == 'x' || format == 'X');
+            // Get the bytes that make up the BigInteger.
+            Span<byte> bits = stackalloc byte[64]; // arbitrary limit to switch from stack to heap
+            bits = value.TryWriteBytes(bits, out int bytesWritten) ?
+                bits.Slice(0, bytesWritten) :
+                value.ToByteArray();
```

Comment:
Not bothering with the array pool?

Reply:
I'll add a commit to use the pool for that.